In [37]:
v , a = confint_proportions_2indep(count1 = yes_f, nobs1 = n_f, count2 = yes_m, nobs2 = n_m,  compare='diff',
                                alpha=0.05, correction = True)
print(v, a)

0.14304339032179628 0.19002232634835703


## Two-Proportion z-interval

In [39]:
# Import the data
titanic = pd.read_csv("titanic_2020.txt", sep = "\t")

In [40]:
# Look at the first few rows
titanic.head()

Name    Born    Died   Age  Gender  \
0                            ABBING, Mr Anthony  1870.0  1912.0  41.0    Male   
1                      ABBOTT, Mr Eugene Joseph  1899.0  1912.0  13.0    Male   
2                    ABBOTT, Mr Rossmore Edward  1896.0  1912.0  16.0    Male   
3                 ABBOTT, Mrs Rhoda Mary 'Rosa'  1873.0  1946.0  39.0  Female   
4  ABELSETH, Miss Kalle (Karen) Marie Kristiane  1895.0  1969.0  16.0  Female   

            Class.Dept Class  Ticket                Fare Cabin  ...  \
0  3rd Class Passenger   3rd    5547  £7             11s   NaN  ...   
1  3rd Class Passenger   3rd  CA2673  £20             5s   NaN  ...   
2  3rd Class Passenger   3rd  CA2673  £20             5s   NaN  ...   
3  3rd Class Passenger   3rd  CA2673  £20             5s   NaN  ...   
4  3rd Class Passenger   3rd  348125  £7             13s   NaN  ...   

   Occupation Survived Boat   Body  Nationality Pounds  Shillings  Pence  \
0  Blacksmith     LOST  NaN    NaN     American    7.0       11.0    NaN   
1     Scholar     LOST  NaN    NaN     American   20.0        5.0    NaN   
2    Jeweller     LOST  NaN  190.0          NaN   20.0        5.0    NaN   
3         NaN    SAVED    A    NaN      English   20.0        5.0    NaN   
4         NaN    SAVED   16    NaN          NaN    7.0       13.0    NaN   

   Price.Paid  Paidpp  
0        7.55    7.55  
1       20.25    6.75  
2       20.25    6.75  
3       20.25    6.75  
4        7.65    7.65  

[5 rows x 21 columns]

In [44]:
pd.crosstab(titanic["Gender"], titanic["Survived"], margins = True)

Survived  LOST  SAVED   All
Gender                     
Female     130    359   489
Male      1366    353  1719
All       1496    712  2208

In [47]:
lower , upper = confint_proportions_2indep(count1 = 359, nobs1 = 489, count2 = 353, nobs2 = 1719,  compare='diff',
                                alpha=0.05, correction = True)
print(lower, upper)

0.4834092767839217 0.5703331209312256


We are 95% confident that the proportion of female who survided in the Titanic is between 48.34% and 57% higher than the proportion of male who survived.

In [86]:
m_f = 39667.2
m_m =  46484.0
sd_f = 37125.9
sd_m = 38699.8
n_f =  465
n_m = 1310

avg_diff= m_m - m_f

se_f = sd_f**2/n_f
se_m = sd_m**2/n_m
se_diff = sqrt(se_f + se_m)
print(se_diff)

2026.676787401113


In [76]:
_tstat_generic(female, male, -1573.9, 1773, alternative = "two-sided")

(4.331151915623612, 1.566296274045779e-05)

In [77]:
465 + 1310

1775

In [79]:
t.interval(0.95, df = 846, loc = avg_diff, scale = se_diff)

(2838.8954767433183, 10794.704523256687)

**Statistics Reminder:**

Depending on whether we have a pooled or non-pooled sample, we will be calculating the degrees of freedom in two different ways: 

* Pooled: you will be doing n (sample size) - 1. In the case of two samples you do: n1 + n2 - 2

* Non-pooled: the formula is quite long, but we can use a function that we have created for you!
    

In [113]:
def degreesOfFreedom(m_1, m_2, sd_1, sd_2, n_1, n_2):
    df = (((sd_1**2)/n_1 + (sd_2**2)/n_2)**2)/ (((sd_1**2)/n_1)**2/(n_1-1) + ((sd_2**2)/n_2)**2/(n_2-1))
    return df

In [114]:
degreesOfFreedom(39667.2, 46484.0,37125.9, 38699.8, 465, 1310)

846.3234163506484